OpenLIFUSonicationPlanner/OpenLIFUSonicationPlanner.py

```diff
@@ -280,10 +396,52 @@ def getParameterNode(self):
         return OpenLIFUSonicationPlannerParameterNode(super().getParameterNode())
 
     def runPlanning(self, inputVolume: vtkMRMLScalarVolumeNode, inputTarget: vtkMRMLMarkupsFiducialNode, inputTransducer : SlicerOpenLIFUTransducer , inputProtocol: SlicerOpenLIFUProtocol ):
-        print("Volume:", inputVolume)
-        print("Target:", inputTarget)
-        print("Protocol:", inputProtocol)
-        print("Transducer:", inputTransducer)
+        plan_info, pnp_aggregated, intensity_aggregated = generate_plan_openlifu(
+            inputProtocol.protocol,
+            inputTransducer,
+            inputTarget,
+            inputVolume,
+        )
+        pnp_volume_node = make_volume_from_xarray_in_transducer_coords(pnp_aggregated, inputTransducer)
+        intensity_volume_node = make_volume_from_xarray_in_transducer_coords(intensity_aggregated, inputTransducer)
+
+        pnp_volume_node.GetDisplayNode().SetAndObserveColorNodeID("vtkMRMLColorTableNodeFilePlasma.txt")
+        intensity_volume_node.GetDisplayNode().SetAndObserveColorNodeID("vtkMRMLColorTableNodeFilePlasma.txt")
+
+        plan = SlicerOpenLIFUPlan(plan_info,pnp_volume_node,intensity_volume_node)
+        slicer.util.getModuleLogic('OpenLIFUData').set_plan(plan)
+
+    def get_pnp(self) -> vtkMRMLScalarVolumeNode:
+        """Get the PNP volume of the active plan, if there is an active plan. Raise exception if there isn't."""
+        plan : SlicerOpenLIFUPlan = slicer.util.getModuleLogic('OpenLIFUData').getParameterNode().loaded_plan
+        if plan is None:
+            raise RuntimeError("Cannot render PNP as there is no active plan.")
```

Comment:
Minor point but it doesn't seem like this check is really needed here since the checkbox is disabled when loaded_plan is empty. So this function would not be called if there were no loaded plans.

Reply:
Maybe still good to have just in case? E.g. in the future this function gets misused and called when it shouldn't, or we introduce a bug where we forget to update the checkbox

Reply:
Yeah no harm in having it

Reply:
Don't forget to approve the PR if you approve :grinning: